lib/api/include/irods/get_resource_info_for_operation.h

```diff
@@ -0,0 +1,38 @@
+#ifndef IRODS_GET_RESOURCE_INFO_FOR_OPERATION_H
+#define IRODS_GET_RESOURCE_INFO_FOR_OPERATION_H
+/// \file
+
+#include "irods/dataObjInpOut.h"
+
+struct RcComm;
+
+/// Get preferred-resource information for the specified operation.
+///
+/// \parblock
+/// \param[in] _conn - An rcComm_t connection handle to the server.
+/// \param[in] _dataObjInp - generic dataObj input. Relevant items are:
+///      \n      \b objPath - the path of the target data object.
+///      \n      \b condInput - conditional Input
+///      \n         - REPL_NUM_KW  - The replica number of the copy to upload.
+///      \n         - GET_RESOURCE_INFO_FOR_OPERATION_KW - The *_OPERATION string: "CREATE", "WRITE", "OPEN", or "UNLINK".
+///      \n         - RESC_NAME_KW - The default destination resource. Only used
+///                                  to create a new file, no overwrite of existing files.
+/// \param[out] _out_info - a JSON string with keys "host" and "resc_hier", identifying target server and resource hierarchy.
+///                         Example:
+///                         \code{.js}
+///                         {
+///                             "host": <string>,
+///                             "resc_hier": <string>
+///                         }
+///                         \endcode
+/// \endparblock
+/// \return integer
+/// \retval 0 on success.
+/// \since 4.3.1
```

Comment:
Add empty line above this one to give unrelated information some room to breathe.

Reply:
Not sure which of the four lines listed above, that you meant should be preceded by a blank-line?

Reply:
Add an empty line above `\since`